core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/kurtosis_plan_instruction/kurtosis_plan_instruction.go

```diff
@@ -50,41 +54,58 @@ func (builtin *KurtosisPlanInstructionWrapper) CreateBuiltin() func(thread *star
 			return nil, interpretationErr
 		}
 
-		var instructionPulledFromMaskIdx int
-		var instructionPulledFromMaskMaybe *instructions_plan.ScheduledInstruction
+		var scheduledInstructionPulledFromMaskMaybe *instructions_plan.ScheduledInstruction
+		var instructionResolutionStatus enclave_structure.InstructionResolutionStatus
 		if builtin.instructionPlanMask.HasNext() {
-			instructionPulledFromMaskIdx, instructionPulledFromMaskMaybe = builtin.instructionPlanMask.Next()
-			if instructionPulledFromMaskMaybe != nil && instructionPulledFromMaskMaybe.GetInstruction().String() != instructionWrapper.String() {
-				// if the instructions differs, then the mask is invalid
-				builtin.instructionPlanMask.MarkAsInvalid()
-				logrus.Debugf("The instruction number %d in the plan mask did not match the newly interpreter "+
-					"instruction and therefore the plan mask was marked as invalid:\nInstruction from mask - '%s'"+
-					"\nInstruction from interpretation: '%s'",
-					instructionPulledFromMaskIdx,
-					instructionPulledFromMaskMaybe.GetInstruction().String(),
-					instructionWrapper.String())
-				// TODO: we could interrupt the interpretation here, because with an invalid mask the list of
-				//  instruction generated will be invalid anyway. Though we currently don't have a nive way to
-				//  interrupt an interpretation in progress (other than by throwing an error, which would be
-				//  misleading here)
-				//  To properly solve that, I think we should switch to an interactive interpretation where we
-				//  interpret each instruction one after the other, and evaluating the state after each step
+			_, scheduledInstructionPulledFromMaskMaybe = builtin.instructionPlanMask.Next()
+			if scheduledInstructionPulledFromMaskMaybe != nil {
+				instructionResolutionStatus = instructionWrapper.TryResolveWith(scheduledInstructionPulledFromMaskMaybe.GetInstruction(), builtin.enclaveComponents)
+			} else {
+				instructionResolutionStatus = instructionWrapper.TryResolveWith(nil, builtin.enclaveComponents)
 			}
+		} else {
+			instructionResolutionStatus = instructionWrapper.TryResolveWith(nil, builtin.enclaveComponents)
 		}
 
-		if instructionPulledFromMaskMaybe != nil {
-			// If there's a mask for this instruction, add the mask the plan and returned the mask's returned value
-			builtin.instructionsPlan.AddScheduledInstruction(instructionPulledFromMaskMaybe).Executed(true).ImportedFromCurrentEnclavePlan(false)
-			return instructionPulledFromMaskMaybe.GetReturnedValue(), nil
-		} else {
+		switch instructionResolutionStatus {
+		case enclave_structure.InstructionIsEqual:
```

Comment:
maybe we could add a default case, just to prevent failing if a new resolution status is added and it's not on this switch ??